tests/parquet_stats.rs

```diff
@@ -0,0 +1,303 @@
+
+    println!("## Thrift meta");
+
+    for rg in file_metadata.row_groups.iter() {
+        for c in rg.columns.iter() {
+            let metadata = c.meta_data.as_ref().unwrap();
+            let stats = metadata.statistics.as_ref().unwrap();
+
+            println!("Path: {:?}", metadata.path_in_schema);
+
+            if let None = stats.min_value.as_ref() {
+                println!("NO min/max");
+            } else {
+                println!("Has min/max");
+            }
+        }
+    }
+
+    let type_ = parquet::schema::types::from_thrift(file_metadata.schema.as_slice()).unwrap();
+    let schema_descriptor = Arc::new(SchemaDescriptor::new(type_));
+
+    println!();
+
+    println!("## Parquet meta");
+
+    let rg_metas = file_metadata
+        .row_groups
+        .iter()
+        .map(|rg| RowGroupMetaData::from_thrift(schema_descriptor.clone(), rg.clone()).unwrap());
+
+    for rg_meta in rg_metas {
+        for c in rg_meta.columns() {
+            println!("Path: {:?}", c.column_path());
+            let stats = c.statistics().unwrap();
+
+            if stats.has_min_max_set() {
+                println!("Has min/max");
+            } else {
+                println!("NO min/max");
+            }
+        }
+    }
+}
+
+fn example_arrow_schema_from_json() -> Arc<ArrowSchema> {
+    let schema_json = json!(
+    {
+      "fields": [
+        {
+          "name": "some_nested_object",
+          "nullable": true,
+          "type": {
+            "name": "struct"
+          },
+          "children": [
+            {
+              "name": "kafka",
+              "nullable": true,
+              "type": {
+                "name": "struct"
+              },
+              "children": [
+                {
+                  "name": "offset",
+                  "nullable": true,
+                  "type": {
+                    "name": "int",
+                    "bitWidth": 64,
+                    "isSigned": true
+                  },
+                  "children": []
+                },
+                {
+                  "name": "topic",
+                  "nullable": true,
+                  "type": {
+                    "name": "utf8"
+                  },
+                  "children": []
+                },
+                {
+                  "name": "partition",
+                  "nullable": true,
+                  "type": {
+                    "name": "int",
+                    "bitWidth": 32,
+                    "isSigned": true
+                  },
+                  "children": []
+                }
+              ]
+            }
+          ]
+        },
+        {
+          "name": "some_int_array",
+          "nullable": true,
+          "type": {
+            "name": "list"
+          },
+          "children": [
+            {
+              "name": "element",
+              "nullable": false,
```

Comment:
@xianwill changing this to `true` fixes the stats and output. same with `"name": "some_struct_array"` below.

I've opened https://github.com/apache/arrow-rs/issues/244 with more context, and I think I found the issue, so going to work on this.

In the mean time, you can make these fields nullable, so that unblocks your progress.